R/test-files.R

```diff
@@ -373,6 +373,10 @@ local_teardown_env <- function(frame = parent.frame()) {
 #' @export
 find_test_scripts <- function(path, filter = NULL, invert = FALSE, ..., full.names = TRUE, start_first = NULL) {
   files <- dir(path, "^test.*\\.[rR]$", full.names = full.names)
+  if (env_var_is_true("CI") || env_var_is_true("NOT_CRAN")) {
```

Comment:
Possibly there is room for an explicit opt in/out env var that is very specific to this feature, like:

```r
run_dev_tests <- function() {
  if (env_var_is_true("RUN_DEV_TESTS")) {
     return(TRUE)
  }

  if (env_var_is_false("RUN_DEV_TESTS")) {
     return(FALSE)
  }

  env_var_is_true("CI") || env_var_is_true("NOT_CRAN")
}
```